### DISCUSSION TITLE
Import json modules not possible?

### DISCUSSION THREAD

**Discussion by nephix** [swc-project/swc#4863] [Q&A] [answered]
I can't seem to find an equivalent for [`resolveJsonModule`](https://www.typescriptlang.org/tsconfig#resolveJsonModule). When I compile my project with `tsc`, it moves imported `.json` files into the dist folder. 

`swc` doesn't and I get the expected error:

```
$ node dist/src
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module './locales/en/email.json'
```

Any idea if that's possible for swc?

**Comment by kdy1** [comment]
You can copy them

**Reply by nephix** [reply]
I know I can but it's not what I asked

**Comment by mohsen1** [comment]
ran into this problem today. JSON import is becoming a standard behavior in the JS ecosystem. I believe SWC should support it as well.

**Comment by kdy1** [comment, marked as answer]
It's supported

2022년 8월 25일 (목) 오전 6:04, Mohsen Azimi ***@***.***>님이 작성:

**Comment by mohsen1** [comment]
I don't see the JSON file in the output. Am I doing something wrong? I'm using `swc` CLI package with `--out-dir`

**Reply by nephix** [reply]
It's [`--copy-files`](https://swc.rs/docs/usage/cli#--copy-files--d)

**Comment by maxh** [comment]
It is possible to define this option in a configuration file instead of via command line flag?

**Reply by kdy1** [reply]
No, and it'll never be supported

**Reply by famouzkk** [reply]
why

**Reply by drekuru** [reply]
Yeah to say it will never be supported? Without an explanation...

Anyway - in NestJS and in our tsconfigs we have the ability to do `includes:[]` regex based matching, when using `swc` with nestjs together, I can't pass in the `--copy-files` param to the `nest start` command as an option - it will simply get dropped. Which is why it would be very nice to have it be a configuration setting.

**Reply by kdy1** [reply]
Copying files is a something that should be done by something outside of ‘@swc/core’

**Reply by drekuru** [reply]
@kdy1 yet there is an option to use `--copy-files`? seems inconsistent. Either the API supports it in both the config file and cli, or it shouldn't support it at all

**Reply by denysshnurenko** [reply]
@drekuru if you use nestjs like me, try the following configuration, it works for me:
```
// nest-cli.json

"compilerOptions": {
  "builder": {
    "type": "swc",
    "options": {
      "copyFiles": true
    }
  }
}
```

**Comment by MLopezJ** [comment]
I resolved this using the `--copy-files` flag. 

Source: https://stackoverflow.com/questions/70866153/import-json-file-in-ts-and-compile-it-with-swc
